From 1c61ec5a3457047987ca5ae709762482269d0ea4 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Tomi=20Lepp=C3=A4nen?= <tomi.leppanen@jolla.com>
Date: Fri, 16 Nov 2018 10:41:00 +0200
Subject: [PATCH] [aboutsettings] Add source for serial number. Fixes JB#38513

Add /sys/class/android_usb/android0/iSerial as a source for serial
number information. usb-moded sets up this information so it should be
always available.
---
 src/aboutsettings.cpp | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/aboutsettings.cpp b/src/aboutsettings.cpp
index d2b3d65..4b895f1 100644
--- a/src/aboutsettings.cpp
+++ b/src/aboutsettings.cpp
@@ -171,6 +171,8 @@ QString AboutSettings::serial() const
     serialFiles
         // This file is specific to the Jolla Tablet;
         << "/config/serial/serial.txt"
+        // usb-moded sets up the serial number here.
+        << "/sys/class/android_usb/android0/iSerial"
         // Some devices have serialno in this path.
         << "/sys/firmware/devicetree/base/firmware/android/serialno";
 
